hexa/workspaces/tests/test_schema/test_workspace.py

```diff
@@ -815,3 +783,50 @@ def test_generate_workspace_token_non_authorized(self):
             },
             r["data"]["generateWorkspaceToken"],
         )
+
+    def test_invite_workspace_member_external_user(self):
+        import random
+        import string
+        from urllib import parse
+
+        with patch("hexa.workspaces.schema.mutations.TimestampSigner") as mocked_signer:
+            random_string = "".join(random.choices(string.ascii_lowercase, k=10))
+
+            signer = mocked_signer.return_value
+            signer.sign.return_value = random_string
+
+            encoded = base64.b64encode(random_string.encode("utf-8")).decode()
+            user_email = "johndoe@foo.com"
+
+            self.client.force_login(self.USER_WORKSPACE_ADMIN)
+            r = self.run_query(
+                """
+                mutation inviteWorkspaceMember($input: InviteWorkspaceMemberInput!) {
+                    inviteWorkspaceMember(input: $input) {
+                        success
+                        errors
+                    }
+                }
+
+                """,
+                {
+                    "input": {
+                        "workspaceSlug": self.WORKSPACE.slug,
+                        "userEmail": user_email,
+                        "role": WorkspaceMembershipRole.VIEWER,
+                    }
+                },
+            )
+
+            invitation = WorkspaceInvitation.objects.filter(email=user_email).first()
```

Comment:
Since you should only have one invitation, I'd recommend to use `WorkspaceInvitation.objects.get(workspace=self.WORKSPACE, email=user_email)`